src/MongoDB.Driver/MongoClient.cs

```diff
@@ -479,13 +495,22 @@ private IReadWriteBindingHandle CreateReadWriteBinding(IClientSessionHandle sess
             ChangeStreamOptions options)
         {
             return ChangeStreamHelper.CreateChangeStreamOperation(
-                pipeline, 
-                options, 
-                _settings.ReadConcern, 
-                GetMessageEncoderSettings(), 
+                pipeline,
+                options,
+                _settings.ReadConcern,
+                GetMessageEncoderSettings(),
                 _settings.RetryReads);
         }
 
+        /// <inheritdoc />
+        public void Dispose()
```

Comment:
Probably this method is not suitable here?
It was added because otherwise, we have the warning in this class since `LibmongoCryptController` implements `IDisposable`, but `MongoClient` - no.
@rstam , how better it can be resolved?

Reply:
As discussed in Slack we should not have `IMongoClient` implement `IDisposable`.